zerver/lib/users.py

```diff
@@ -65,6 +65,21 @@ def check_full_name(full_name_raw: str) -> str:
     # ban them.
     if re.search(r"\|\d+$", full_name_raw):
         raise JsonableError(_("Invalid format!"))
+
+    if require_unique_names(realm):
+        normalized_user_full_name = unicodedata.normalize("NFKC", full_name)
+        realm_users_names = (
+            UserProfile.objects.filter(realm=realm)
+            .exclude(full_name=full_name)
```

Comment:
No, don't exclude by name! That allows exact duplicates. You probably need to plumb through the current `user_profile` in the code paths for changing one's name.

Reply:
ok yeah, that makes sense...updated!